### PR TITLE
Implement `createTHead` method on `<table>`.

### PR DESCRIPTION
**Pull request by frewsxcv**
Fixes https://github.com/servo/servo/issues/10506.

Part of https://github.com/servo/servo/issues/9023.

Part of https://github.com/servo/servo/issues/9769.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10507)

### CODE REVIEW

**Comment by KiChjang** [comment]
Step annotations would be nice.

**Comment by KiChjang** [comment]
Nevermind, the spec doesn't have any.

File: components/script/dom/htmltableelement.rs [hunk 1]
```diff
@@ -119,6 +120,28 @@ impl HTMLTableElementMethods for HTMLTableElement {
...
+                first_viable = Some(child);
+            }
+        }
+        let thead = HTMLTableSectionElement::new(atom!("thead"),
+                                                 None,
+                                                 document_from_node(self).r());
+        if let Some(node) = first_viable {
+            self.upcast::<Node>().InsertBefore(thead.upcast(), Some(node.upcast::<Node>())).unwrap();
+        } else {
+            self.upcast::<Node>().InsertBefore(thead.upcast(), None).unwrap();
```
Comment (KiChjang) [inline components/script/dom/htmltableelement.rs]: Perhaps `self.upcast::<Node>().AppendChild(thead.upcast()).unwrap();` would make more sense here?

Comment (frewsxcv) [inline components/script/dom/htmltableelement.rs]: Done.

**Comment by frewsxcv** [comment]
Didn't realize https://github.com/servo/servo/pull/9980 existed.

**Pull request closed without merging**